src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/WarningTaskHandlerTest.java

```diff
@@ -1,30 +1,127 @@
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.DmnStringValue;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EvaluateDmnRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EvaluateDmnResponse;
 import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EventInformation;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.ProcessVariables;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.SendMessageRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.warningtask.WarningEvaluateResponse;
 import uk.gov.hmcts.reform.wacaseeventhandler.handlers.WarningTaskHandler;
 
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+@ExtendWith(MockitoExtension.class)
 class WarningTaskHandlerTest {
 
-    private final WarningTaskHandler handlerService = new WarningTaskHandler();
+    private static final String DMN_NAME = "wa-task-cancellation-ia-asylum";
+    public static final String WARN_TASKS_MESSAGE_NAME = "warnProcess";
+
+
+    @Mock
+    private WorkflowApiClientToWarnTask workflowApiClientToWarnTask;
+
+    @Captor
+    private ArgumentCaptor<SendMessageRequest<ProcessVariables, CancellationCorrelationKeys>> sendMessageRequestCaptor;
+
+    @InjectMocks
+    private WarningTaskHandler handlerService;
+
+    private final EventInformation eventInformation = EventInformation.builder()
+        .eventId("some event id")
+        .newStateId("some post state")
+        .previousStateId("some previous state")
+        .jurisdictionId("ia")
+        .caseTypeId("asylum")
+        .caseReference("some case reference")
+        .dateTime(LocalDateTime.now())
+        .build();
 
     @Test
     void evaluateDmn() {
-        assertThat(handlerService.evaluateDmn(
-            EventInformation.builder()
-                .jurisdictionId("ia")
-                .caseTypeId("asylum")
-                .build()))
-            .isEmpty();
+        CancellationEvaluateRequest cancellationTaskEvaluateDmnRequestVariables =
+            CancellationEvaluateRequest.builder()
+                .state(new DmnStringValue("some post state"))
+                .event(new DmnStringValue("some event id"))
+                .fromState(new DmnStringValue("some previous state"))
+                .build();
+
+        EvaluateDmnRequest<CancellationEvaluateRequest> requestParameters =
+            new EvaluateDmnRequest<>(cancellationTaskEvaluateDmnRequestVariables);
+
+        when(workflowApiClientToWarnTask.evaluateDmn(
+            DMN_NAME,
+            requestParameters
+        )).thenReturn(new EvaluateDmnResponse<>(Collections.emptyList()));
```

Comment:
Probably worth having a test where the response is returning values also and then can assert that the same ones are returned.